src/main/webapp/js/searchableField.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Modified from the Jenkins combobox code
+ * https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/form/combobox/combobox.js
+ *
+ * The Jenkins combobox only fills items 'onChange' of dependent fields. This means that if a combobox is dependent on
+ * itself (e.g a search field) then it is filled too late. This will fill the combobox options on 'input' and then
+ * when one is selected the human readable value will be in the combobox input and a computer-readable value will go
+ * in the associated value field.
+ **/
+Behaviour.specify(".searchable", 'searchableField', 200, function(el) {
```

Comment:
AFAICT the `priority` of `200` is largely irrelevant here and we could change it to `0` to indicate we don't care.

The alternative [`onload` method in the page util](https://github.com/jenkinsci/jenkins/blob/22aa2e6e766074d11249893e3f35e0b99e20d3d0/war/src/main/js/util/page.js#L15) is a bit simpler and just requires a selector and callback.

Reply:
As discussed in person, there are some improvements to do with this js but I will address them in another PR